### PR TITLE
#20 API for go: added docs

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #20 added docs for go module proxy protocol and todo for next steps.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#23](https://github.com/artipie/go-adapter/issues/23) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/3db0dc83-ddab-4433-ac75-4f939cceb86d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: README.md [hunk 1]
```diff
@@ -106,6 +106,152 @@ be created when you call `update("example.com/foo/first", "0.0.2").blockingAwait
...
+go tool and `go get` command. Basically, `go get` can get desired modules either from vcs or download
+it over HTTP proxy using module proxy protocol. In Golang docs detailed description about this 
+mechanism can be found in [Remote import path](https://golang.org/cmd/go/#hdr-Remote_import_paths) 
+section. 
+
+As described in [Golang docs](https://golang.org/cmd/go/#hdr-Module_proxy_protocol), go module proxy 
+is any web server that can respond to GET requests of specified form. While working with proxy `go get` 
+obtains all the modules and dependencies using configured proxy sending set of GET requests to it.
+
+Here is a full list of the requests sent along with corresponding responses:
```
Comment (Vatavuk) [inline README.md:121]: @olenagerasimova `Here is a full list of GET requests sent to a Go module proxy with corresponding examples`

File: README.md [hunk 2]
```diff
@@ -106,6 +106,152 @@ be created when you call `update("example.com/foo/first", "0.0.2").blockingAwait
...
+
+The most common way to get any source code, modules or packages while working with go is to use 
+go tool and `go get` command. Basically, `go get` can get desired modules either from vcs or download
+it over HTTP proxy using module proxy protocol. In Golang docs detailed description about this 
+mechanism can be found in [Remote import path](https://golang.org/cmd/go/#hdr-Remote_import_paths) 
+section. 
+
+As described in [Golang docs](https://golang.org/cmd/go/#hdr-Module_proxy_protocol), go module proxy 
+is any web server that can respond to GET requests of specified form. While working with proxy `go get` 
+obtains all the modules and dependencies using configured proxy sending set of GET requests to it.
```
Comment (Vatavuk) [inline README.md:119]: @olenagerasimova I believe this sentence is redundant.

**Comment by olenagerasimova** [comment]
@Vatavuk thanks, corrected, take another look, please

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure that during next CR you would NOT only find cosmetic issues, but rather design problems

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**